src/components/ContentSidebar/ActivityFeed/Avatar.js

```diff
@@ -35,22 +35,24 @@ class Avatar extends React.PureComponent<Props, State> {
     /**
      * Gets the avatar URL for the user from the getAvatarUrl prop
      *
-     * @return {Promise} a promise which resolves with the avatarUrl string
+     * @return {Promise<?string>} Promise which resolve with the avatar url string
      */
-    getAvatarUrl() {
+    getAvatarUrl(): Promise<?string> {
         const { user, getAvatarUrl }: Props = this.props;
-        return getAvatarUrl(user.id).then(this.getAvatarUrlHandler);
+
+        const avatarPromise = getAvatarUrl ? getAvatarUrl(user.id) : Promise.resolve(user.avatar_url);
+        return avatarPromise.then(this.getAvatarUrlHandler);
     }
 
     componentDidMount() {
         this.getAvatarUrl();
     }
 
     render() {
-        const { user, className }: Props = this.props;
+        const { user, className, getAvatarUrl }: Props = this.props;
         const { avatarUrl }: State = this.state;
 
-        if (!avatarUrl) {
+        if (getAvatarUrl && !avatarUrl) {
```

Comment:
What's the reason for adding a default behavior for `getAvatarUrl` if we avoid rendering when the prop isn't defined?

Reply:
I can actually remove this check altogether. If no avatar url is defined, it should still render the component just w/ teh initials instead. That is handled inside the AvatarComponent